frontend/public/components/monitoring.tsx

```diff
@@ -159,24 +159,39 @@ const SilenceMatchersList = ({silence}) => <div className={`co-text-${SilenceRes
   {_.map(silence.matchers, ({name, isRegex, value}) => <Label key={name} k={name} v={isRegex ? `~${value}` : value} />)}
 </div>;
 
+const pollerLastCallTimes = {};
+
 class MonitoringPageWrapper extends SafetyFirst<AlertsPageWrapperProps, null> {
   componentDidMount () {
     super.componentDidMount();
 
+    const interval = 15 * 1000;
+    const now = () => (new Date()).getTime();
+
+    const safeSetTimeout = (cb, timeout) => setTimeout(() => {
+      if (this.isMounted_) {
```

Comment:
Not against this, but I wonder if it's better to cancel the timeout on unmount instead

Reply:
👍Changed to cancel the timeouts in `componentWillUnmount()`